agent/lib/kontena/workers/container_health_check_worker.rb

```diff
@@ -78,10 +76,17 @@ def check_http_status(ip, port, uri, timeout)
           connect_timeout: timeout,
           headers: {
             "User-Agent" => "Kontena-Agent/#{Kontena::Agent::VERSION}"
-          }
+          },
+          middlewares: [
+            Excon::Middleware::ResponseParser,
+            Excon::Middleware::Expects,
+            Excon::Middleware::Idempotent,
+            Excon::Middleware::Instrumentor,
+            Excon::Middleware::Mock,
+          ]
         )
         debug "got status: #{response.status}"
-        data['status'] = HEALTHY_STATUSES.include?(response.status) ? 'healthy' : 'unhealthy'
+        data['status'] = (response.status >= 200 && response.status < 400) ? 'healthy' : 'unhealthy'
```

Comment:
How about `(200..399).cover?(response.status)` or `response.status.between?(200, 399)`

Reply:
I considered `HEALTHY_STATUSES = (200...400)` and `HEALTHY_STATUSES.include?(response.status)`, but the `...` vs `..` is way too subtle to actually use. I would vaguely prefer the more explicit use of `>=` and `<` for the open vs closed range... there's many different ways to write this condition.

Reply:
Reverted this to just keep using `HEALTHY_STATUSES.include?`... but got rid of the unnecessary `to_a` which starts to hurt me now that the resulting array actually contains ~200 entries.

Reply:
`Range#cover?` does not `.to_a`, `Comparable#between?` neither.

Reply:
The existing code does, but I don't think there's any good reason for it to do so: https://github.com/kontena/kontena/blob/e95dc72104e7207a8dd70e6b31ce4666f7682aa4/agent/lib/kontena/workers/container_health_check_worker.rb#L12